src/frontend/components/app/modal-info.tsx

```diff
@@ -0,0 +1,43 @@
+import { Modal } from '@adminjs/design-system'
+import React from 'react'
+import { useSelector } from 'react-redux'
+import { ModalButton } from '../../interfaces'
+import { ReduxState } from '../../store'
+import { useModal, useTranslation } from '../../hooks'
+
+export const ModalInfo = () => {
+  const rawModal = useSelector((state: ReduxState) => state.modal)
+  const { closeModal } = useModal()
+  const { translateButton, translateLabel, translateMessage } = useTranslation()
+  const modal = { ...rawModal }
```

Comment:
Why is spread needed here?

Reply:
because I have to change some data from state before I pass them to modal component